File: Flutter App/lib/models/customer_auth_response.dart
import 'customer_profile.dart';

class CustomerAuthResponse {
  final bool success;
  final String? message;
  final String? token;
  final String? role;
  final CustomerProfile? customer;

  const CustomerAuthResponse({
    required this.success,
    this.message,
    this.token,
    this.role,
    this.customer,
  });

  factory CustomerAuthResponse.fromJson(Map<String, dynamic> json) {
    final customerJson = json['customer'];
    return CustomerAuthResponse(
      success: json['success'] == true,
      message: json['message']?.toString(),
      token: json['token']?.toString(),
      role: json['role']?.toString(),
      customer: customerJson is Map<String, dynamic>
          ? CustomerProfile.fromJson(customerJson)
          : null,
    );
  }
}

File: Backend - Spring Boot/src/main/java/com/styleora/dto/CustomerAuthResponse.java
package com.styleora.dto;

public class CustomerAuthResponse {

    private boolean success;
    private String message;
    private String token;
    private String role;
    private CustomerProfileResponse customer;

    public CustomerAuthResponse() {
    }

    public CustomerAuthResponse(boolean success, String message, String token, String role, CustomerProfileResponse customer) {
        this.success = success;
        this.message = message;
        this.token = token;
        this.role = role;
        this.customer = customer;
    }

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public CustomerProfileResponse getCustomer() {
        return customer;
    }

    public void setCustomer(CustomerProfileResponse customer) {
        this.customer = customer;
    }
}
